pkg/kubectl/cmd/portforward.go

```diff
@@ -76,11 +75,21 @@ func (*defaultPortForwarder) ForwardPorts(req *client.Request, config *client.Co
 
 func RunPortForward(f *cmdutil.Factory, cmd *cobra.Command, args []string, fw portForwarder) error {
 	podName := cmdutil.GetFlagString(cmd, "pod")
-	if len(podName) == 0 {
-		return cmdutil.UsageError(cmd, "POD_NAME is required for port-forward")
+	if len(podName) == 0 && len(args) == 0 {
+		return cmdutil.UsageError(cmd, "POD is required for port-forward")
 	}
-	if len(args) < 1 {
-		return cmdutil.UsageError(cmd, "at least 1 PORT is required for port-forward")
+	if len(podName) != 0 {
+		printDeprecationWarning("port-forward POD", "-p POD")
+		if len(args) < 1 {
+			return cmdutil.UsageError(cmd, "at least 1 PORT is required for port-forward")
+		}
+
+	} else {
+		podName = args[0]
+		if len(args) < 2 {
```

Comment:
Remove the first arg from the list, then move this check below the if/else to unify both paths.  Otherwise it's easy for the messages to drift.